src/iologindata.cpp

```diff
@@ -776,6 +826,33 @@ bool IOLoginData::savePlayer(Player* player)
 		}
 	}
 
+	auto rewardList = player->getRewardList();
```

Comment:
Move this to line 834 !db->executeQuery(query.str()
Waste to load it if the query statment returns false.